realm/realm-annotations-processor/src/test/java/io/realm/processor/RealmProcessorTest.java

```diff
@@ -479,6 +480,36 @@ public void compileWithInterfaceForObject() {
     }
 
     @Test
+    public void compileCounter() {
+        // FIXME Counters: Add test
+    }
+
+    @Test
+    public void compileStaticCounter() {
+        // FIXME Counters: Add test
+    }
+
+    @Test
+    public void compileIgnoredCounter() {
+        // FIXME Counters: Add test
+    }
+
+    @Test
+    public void compileRequiredCounter() {
+        // FIXME Counters: Add test
+    }
+
+    @Test
+    public void failOnPKCounter() {
+        // FIXME Counters: Add test
+    }
+
+    @Test
+    public void failOnIndexedCounter() {
```

Comment:
This is not in the spec.  Yes, I know that is because no spec exists.

Ok.  Will add to the spec.  Allowing both @Required and @Index seem like good choices.